beacon-chain/db/filesystem/save_blob.go

```diff
@@ -0,0 +1,88 @@
+// The filesystem package provides a mechanism for managing and storing binary blob data.
+// It offers functionalities for saving and ensuring the integrity of blobs using SHA-256 checksums.
+// In order to maintain data integrity we first write to a temporary file and then atomically
+// rename the file to its final destination.
+
+package filesystem
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/io/file"
+	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+)
+
+type BlobStorage struct {
+	baseDir string
+}
+
+// SaveBlobData saves blobs given a list of sidecars.
+func (bs *BlobStorage) SaveBlobData(sidecars []*ethpb.BlobSidecar) error {
+	if len(sidecars) == 0 {
+		return errors.New("no blob data to save")
+	}
+	for _, sidecar := range sidecars {
+		blobPath := path.Join(bs.baseDir, fmt.Sprintf(
+			"%d_%x_%d_%x.blob",
+			sidecar.Slot,
+			sidecar.BlockRoot,
+			sidecar.Index,
+			sidecar.KzgCommitment,
+		))
+		exists := file.Exists(blobPath)
+		if exists {
+			if err := checkDataIntegrity(sidecar.Blob, blobPath); err != nil {
+				return errors.Wrapf(err, "failed to save blob sidecar, tried to overwrite blob (%s) with different content", blobPath)
+			}
+			continue // Blob already exists, move to the next one
+		}
+		// Create a partial file and write the blob data to it.
+		partialFilePath := blobPath + ".partial"
+		partialFile, err := os.Create(partialFilePath)
+		if err != nil {
+			return errors.Wrap(err, "failed to create partial file")
+		}
+
+		_, err = partialFile.Write(sidecar.Blob)
```

Comment:
Note, it might be easier to save the whole `sidecar`. Or you will have to manually compute kzg proof and inclusion proofs later on if someone requests it

Reply:
Good feedback. We should discuss what format to store the blob sidecar. Probably ssz?